include/libcyphal/presentation/errors.hpp

```diff
@@ -0,0 +1,53 @@
+/// @copyright
+/// Copyright (C) OpenCyphal Development Team  <opencyphal.org>
+/// Copyright Amazon.com Inc. or its affiliates.
+/// SPDX-License-Identifier: MIT
+
+#ifndef LIBCYPHAL_PRESENTATION_ERRORS_HPP_INCLUDED
+#define LIBCYPHAL_PRESENTATION_ERRORS_HPP_INCLUDED
+
+#include "libcyphal/errors.hpp"
+#include "libcyphal/types.hpp"
+
+#include <cetl/pf17/cetlpf.hpp>
+
+#include <nunavut/support/serialization.hpp>
+
+namespace libcyphal
+{
+namespace presentation
+{
+
+/// @brief Defines terminal 'expired' error state of the response promise.
```

Comment:
Let us please keep these highly specific error states next to their place of use. I think the promise-related errors should go right next to the Client definition, as they effectively form its API.

Reply:
No, I meant that they should be next to the Client class definition but not part of it

Reply:
ok, I can do it, although I have 2 points more:
1. strictly speaking these two are more related to promise rather than client - hence the "ResponsePromised" prefix
2. there is kinda pattern at other layers of libcyphal, namely each layer has their own `<layer>/errors.hpp` which holds definitions of free standing errors and failures (variants).

Reply:
So are we moving them next to the promise? I get the idea of moving generic error types like argument error or memory error into an errors.hpp, but more specialized types should be kept close to the entity they pertain to.

Reply:
done